File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    
    const days = new Array(progresses.length).fill(0);
    for (let d=0; d<days.length; d++){
        days[d] = Math.floor((100-progresses[d]+speeds[d]-1)/speeds[d]);
    }
    while (days.length>0){
        const top = days.shift();
        let cnt = 1;
        while (days.length>0 && days[0]<=top){
            days.shift();
            cnt++;
        }
        answer.push(cnt);
    }
    
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    static class Work {
        int progress, speed;
        Work(int progress, int speed){
            this.progress = progress;
            this.speed = speed;
        }
        void addProgress(){
            if (this.progress<100) this.progress +=speed;
        }
    }
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Work> queue = new LinkedList<>();
        
        for (int i=0; i<progresses.length; i++){
            queue.offer(new Work(progresses[i], speeds[i]));
        }
        
        List<Integer> resultList = new ArrayList<>();
        
        while (!queue.isEmpty()){
            boolean release = true;
            int len = queue.size();
            int works = 0;
            for (int i=0; i<len; i++){
                Work work = queue.poll();
                work.addProgress();
                if (work.progress>=100 && release){
                    works++;
                } else {
                    release = false;
                    queue.offer(work);
                }
            }
            if (works>0) resultList.add(works);
        }
        
        int[] answer = new int[resultList.size()];
        for (int i=0; i<answer.length; i++){
            answer[i] = resultList.get(i);
        }
        
        return answer;
    }
}